docs(installation): prioritize official installer and recommends nvs/nvm (#1144)
hexojs/site: source/docs/index.md

If not, please follow the following instructions to install all the requirements.

### Install Git

- Windows: Download & install [git](https://git-scm.com/download/win).
- Mac: Install it with [Homebrew](https://brew.sh/), [MacPorts](http://www.macports.org/) or [installer](http://sourceforge.net/projects/git-osx-installer/).
- Linux (Ubuntu, Debian): `sudo apt-get install git-core`
- Linux (Fedora, Red Hat, CentOS): `sudo yum install git-core`

{% note warn For Mac users %}
You may encounter some problems when compiling. Please install Xcode from App Store first. After Xcode is installed, open Xcode and go to **Preferences -> Download -> Command Line Tools -> Install** to install command line tools.
{% endnote %}

### Install Node.js

> Node.js provides [official installer](https://nodejs.org/en/download/) for most platforms.

Reviewer: ```
{% note info For Windows users %}
When install Node.js using official installer, make sure **Add to PATH** is checked.
{% endnote %}
```

An `Add to PATH` notice for Windows.

Alternative installation methods:

> - Windows: Install it with [nvs](https://github.com/jasongin/nvs/) (recommended), [nvm](https://github.com/nvm-sh/nvm) or [Chocolatey](https://chocolatey.org/). (nvs is also recommended for Mac and Linux to avoid possible permission issue)

Reviewer: ```suggestion
- Windows: Install it with [nvs](https://github.com/jasongin/nvs/) (recommended), [nvm](https://github.com/nvm-sh/nvm) or [Chocolatey](https://chocolatey.org/).
```

- Mac: Install it with [Homebrew](https://brew.sh/) or [MacPorts](http://www.macports.org/).
- Linux (DEB/RPM-based): Install it with [NodeSource](https://github.com/nodesource/distributions).

> - Linux (other): Install it through respective package manager (if Node.js >= 10.0 is available) or use nvs/nvm.

Reviewer: ```suggestion
- Linux (other): Install it through respective package manager (if Node.js >= 10.0 is available).

[nvs](https://github.com/jasongin/nvs/) and [nvm](https://github.com/nvm-sh/nvm) are also recommended for Mac and Linux.
```

The recommendation for Mac & Linux should be here.

{% note warn Mac / Linux %}

> If you encounter permission error when trying to install Hexo, please follow the workaround [provided by npmjs](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally); overriding with root/sudo is highly discouraged.

Reviewer: ```suggestion
If you encounter `EACCES` error when when trying to install Hexo. Please follow [the workaround provided by npmjs](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally); overriding with root/sudo is highly discouraged.
```

Also this notice might need to be placed under `Install Hexo` section.
